### PR TITLE
feat(docker): add support for URL environment variable

### LINKED ISSUE #2878: Docker deployments: add support for the URL environment variable
### Content & configuration

Perform a Docker deployment of the swagger-editor image with the following options:

 ```sh
docker run -e URL="https://petstore3.swagger.io/api/v3/openapi.json" -d -p 9999:8080 swaggerapi/swagger-editor
 ```
Then, simply browse to http://localhost:9999 and check the "home" API document loaded by default (Swagger Petstore). 

### Is your feature request related to a problem?

Yes. With current Docker image, swagger-editor ignores the `URL` environment variable and always provide the same default API document on startup.

 
### Describe the solution you'd like

Some swagger-ui environment variables are already supported by swagger-editor (e.g., `BASE_URL`, `QUERY_CONFIG_ENABLED`, etc.) .
I would like to propose a PR to support the `URL` environment variable, in line with swagger-ui configuration/reference (see below). 

For the users who deploy swagger-editor with Docker, this change would allow them to replace the default API document to be loaded on startup with an URL (see after for a discussion on the `LocalStorage` plugin). 

### Describe alternatives you've considered

An alternative exists but it lacks of flexibility and allows the users to override configuration parameters via URL search params themselves.

In this alternative, the swagger-editor image has to be deployed with the `QUERY_CONFIG_ENABLED` environment variable:

 ```sh
docker run -e QUERY_CONFIG_ENABLED="true" -d -p 9999:8080 swaggerapi/swagger-editor
 ```
Then, the homepage would be: http://localhost:9999/?url=https://petstore3.swagger.io/api/v3/openapi.json

However, if we then want to display another API document on startup, we are forced to update the link to be shared, with the risk that some users could still land on the "old" API document/homepage. 

### Additional context

I already investigated a bit the changes to be done. The following files would require some changes:

- `docker-run.sh`
- `index.html`
- `README.md` (documentation)
- `LocalStorage` plugin (see below) 

Some changes could be required for the `LocalStorage` plugin so that it does not first load the Swagger Petstore default document when the `URL` environment variable is specified by the user (to remove the "blinking" effet).

I have a first PoC working on my swagger-editor fork. However, because I first wanted to make something simple, I didn't change the logic of the `LocalStorage` plugin yet. Of course, I could continue to work on it if you think that it makes sense.

[Swagger-UI documentation](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md) for reference and alignement:
Method name | Docker variable | Description
--- | --- | -----
<a name="url"></a>`url` | `URL` | `String`. The URL pointing to API definition (normally `swagger.json` or `swagger.yaml`). Will be ignored if `urls` or `spec` is used.

### PR DESCRIPTION
**Pull request by antoineauger**
### Description
For Docker-based deployments, this PR allows users to specify a default remote API document to be loaded on startup (when the local cache is empty).

### Motivation and Context
Until now, swagger-editor was ignoring the `URL` environment variable, always providing the same default API document on startup (Swagger Petstore). Users could still set `SWAGGER_FILE` env variable but this was only working with local file(s).

This PR aims at allowing users to specify an alternative remote API document to be loaded on startup for Docker-based deployments:

```sh
docker run -d -p 80:8080 -e URL="https://petstore3.swagger.io/api/v3/openapi.json" swaggerapi/swagger-editor
```

When both `URL` and `SWAGGER_FILE` env variables are set, `URL` has priority and `SWAGGER_FILE` is ignored.

Fixes #2878 

### How Has This Been Tested?

Tested locally with the following variable combinations:

- `URL` env variable set => API document fetched from URL

Content of `index.html` after startup:

```js
window.onload = function() {
    // Build a system
    const editor = SwaggerEditorBundle({
      url: 'https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v3.0/uspto.json',
      dom_id: '#swagger-editor',
      layout: 'StandaloneLayout',
      presets: [
        SwaggerEditorStandalonePreset
      ],
      queryConfigEnabled: false,
    })

    window.editor = editor
  }
```

- `SWAGGER_FILE` + `URL` env variables set => API document fetched from URL

Content of `index.html` after startup:

```js
  window.onload = function() {
    // Build a system
    const editor = SwaggerEditorBundle({
      url: 'https://raw.githubusercontent.com/OAI/OpenAPI-Specification/main/examples/v3.0/uspto.json',
      dom_id: '#swagger-editor',
      layout: 'StandaloneLayout',
      presets: [
        SwaggerEditorStandalonePreset
      ],
      queryConfigEnabled: false,
    })

    window.editor = editor
  }
```

- `SWAGGER_FILE` env variable set => API document fetched locally

Content of `index.html` after startup:

```js
  window.onload = function() {
    // Build a system
    const editor = SwaggerEditorBundle({
      url: '/ex.json',
      dom_id: '#swagger-editor',
      layout: 'StandaloneLayout',
      presets: [
        SwaggerEditorStandalonePreset
      ],
      queryConfigEnabled: false,
    })

    window.editor = editor
  }
```

- no env variables set => default Swagger Petstore API document displayed

```js
  window.onload = function() {
    // Build a system
    const editor = SwaggerEditorBundle({
      dom_id: '#swagger-editor',
      layout: 'StandaloneLayout',
      presets: [
        SwaggerEditorStandalonePreset
      ],
      queryConfigEnabled: false,
    })

    window.editor = editor
  }
```

### Screenshots (if appropriate):
N/A

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [ ] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [x] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [ ] My changes do not require a change to the project documentation.
- [x] My changes require a change to the project documentation.
- [x] If yes to above: I have updated the documentation accordingly.

### Automated tests
- [x] My changes can not or do not need to be tested.
- [ ] My changes can and should be tested by unit and/or integration tests.
- [ ] If yes to above: I have added tests to cover my changes.
- [ ] If yes to above: I have taken care to cover edge cases in my tests.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,19 +113,20 @@ docker run -d -p 80:8080 swaggerapi/swagger-editor
 
 This will run Swagger Editor (in detached mode) on port 80 on your machine, so you can open it by navigating to `http://localhost` in your browser.  
 
-
-* You can provide your own `json` or `yaml` definition file on your host
+* You can provide the link of a remote API document to be loaded by the editor on startup (may not be available if some security policies such as CSP or CORS are enforced):
```
Comment (char0n) [inline README.md:116]:
```suggestion
* You can provide a URL pointing to an API definition (may not be available if some security policies such as CSP or CORS are enforced):
```

**Comment by char0n** [comment]
Looks good. According the the tests results you provided the new functionality works and we have full backward compatibility as well.

**Comment by antoineauger** [comment]
> Looks good. According the the tests results you provided the new functionality works and we have full backward compatibility as well.

Just provided additional results for the test cases performed.

**Review by char0n — APPROVED** [review]
Thanks for contributing! I'll run the local tests as well before I'll merge and issue a new release.

**Comment by dlouzan** [comment]
Great work, thanks you both @char0n and @antoineauger 💯

**Comment by galvo** [comment]
I just pulled down the latest version and i used to be able to pass '?url=remotefile.yaml' in  v4.0.0 but that no longer seems to work now?

**Comment by antoineauger** [comment]
> I just pulled down the latest version and i used to be able to pass '?url=remotefile.yaml' in v4.0.0 but that no longer seems to work now?

Hi @galvo! I just tested it and I don't have this issue with latest version...

Some things you may want to check:
- Did you start your container with the env variable `QUERY_CONFIG_ENABLED="true"`? This is required in your case (see https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md)
- If your `remotefile.yaml` is actually a file inside a directory mounted with `-v` command, the target path has to be in nginx `html` directory (e.g., `-v /bar:/usr/share/nginx/html/foo`, then use `?url=/foo/remotefile.yaml` in url when the container is running).
- If it is a "true" remote file (`https://...`) and the url is correct, you could have some security policies and/or network configuration that prevent swagger-editor from fetching the file.

If the problem still persists, feel free to open a bug issue with more debug details.
